pkg/ibmvpc/explainability_test.go

```diff
@@ -738,3 +739,62 @@ func TestInputValidityMultipleVPCContext(t *testing.T) {
 		"- test-vpc0-ky, test-vpc1-ky - please add the name of the config to the src/dst name",
 		err11.Error())
 }
+
+// sanity check: no error and expected number of explanation elements
+func TestMultiExplainSanity(t *testing.T) {
+	vpcsConfig := getConfig(t, "tgw_larger_example")
+	require.NotNil(t, vpcsConfig, "vpcsConfig equals nil")
+	groupedConns := make(map[string]*vpcmodel.GroupConnLines)
+	nodesConn := make(map[string]*vpcmodel.VPCConnectivity)
+	for i := range vpcsConfig {
+		thisConn, err := vpcsConfig[i].GetVPCNetworkConnectivity(false)
+		if err != nil {
+			fmt.Printf("%v. %s", i, err.Error())
+		}
+		require.Nil(t, err)
+		groupedConns[i] = thisConn.GroupedConnectivity
+		nodesConn[i] = thisConn
+	}
+	inputMultiExplain := vpcmodel.CreateMultiExplanationsInput(vpcsConfig, groupedConns)
+	multiExplain := vpcmodel.MultiExplain(inputMultiExplain, nodesConn)
+	i := 0
+	for _, explain := range multiExplain {
+		require.Equal(t, "", explain.Error())
+		i++
+	}
+	require.Equal(t, i, len(inputMultiExplain))
+}
+
+func TestMultiExplainabilityOutput(t *testing.T) {
+	vpcsConfig := getConfig(t, "tgw_basic_example")
+	require.NotNil(t, vpcsConfig, "vpcsConfig equals nil")
+	groupedConns := make(map[string]*vpcmodel.GroupConnLines)
+	nodesConn := make(map[string]*vpcmodel.VPCConnectivity)
+	for i := range vpcsConfig {
+		thisConn, err := vpcsConfig[i].GetVPCNetworkConnectivity(false)
+		if err != nil {
+			fmt.Printf("%v. %s", i, err.Error())
+		}
+		require.Nil(t, err)
+		groupedConns[i] = thisConn.GroupedConnectivity
+		nodesConn[i] = thisConn
+	}
+	inputMultiExplain := vpcmodel.CreateMultiExplanationsInput(vpcsConfig, groupedConns)
+	multiExplain := vpcmodel.MultiExplain(inputMultiExplain, nodesConn)
+	outputSlice := make([]string, len(multiExplain))
+	for i, explain := range multiExplain {
+		require.Equal(t, "", explain.Error())
+		outputSlice[i] = explain.String()
+	}
+	outputString := strings.Join(outputSlice, "")
+	require.Contains(t, outputString, "No connections are allowed from Public Internet (all ranges) to ky-vpc2-vsi[10.240.64.5];\n"+
+		"no fip and src is external (fip is required for outbound external connection)", "no connection external src entry")
+	require.Contains(t, outputString, "No connections are allowed from ky-vpc2-vsi[10.240.64.5] to Public Internet (all ranges);\n"+
+		"no fip/pgw and dst is external", "no connection external dst entry")
+	require.Contains(t, outputString, "ky-vpc1-vsi[10.240.0.5] -> security group ky-vpc1-sg -> ky-vpc1-net1 -> network ACL ky-vpc1-acl1 ->",
+		"connection vsi to vsi")
+	require.Contains(t, outputString, "ky-vpc1 -> TGW local-tg-ky -> ky-vpc2 ->", "connection vsi to vsi")
+	require.Contains(t, outputString, "network ACL ky-vpc2-acl1 -> ky-vpc2-net1 -> security group ky-vpc2-sg -> ky-vpc2-vsi[10.240.64.5]",
+		"connection vsi to vsi")
```

Comment:
can you find alternative to use hard-coded strings here? such as using the generated output files for per-query explainability tests, and check containment of such files contents within the multi-explain output?

Reply:
Can you open a separate issue for this? Also, this issue can include other tests with hard-coded comparsion

Reply:
issue #544